### DISCUSSION TITLE
setup already postgress installed

### DISCUSSION THREAD

**Discussion by joossii** [simple-login/app#958] [Self-hosting question]
Hey,

So I have already Postgress installed and when i use localhost or server ip or localhostip(127...) i keep getting:

 conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: could not connect to server: Connection timed out
        Is the server running on host "localhost or server ip or localhostip(127...)" and accepting
        TCP/IP connections on port 5432?

after running 
docker run --rm \
    --name sl-migration \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    -v $(pwd)/simplelogin.env:/code/.env \
    --network="sl-network" \
    simplelogin/app:3.4.0 flask db upgrade

can someone help me set this up with a postgres outside a docker container?

**Comment by nguyenkims** [comment]
You need to update the SimpleLogin `DB_URI` config to point to your existing database. As SimpleLogin is run as containers, the database needs to be reachable from the docker container.

**Reply by joossii** [reply]
you mean this part:
# DB Connection
DB_URI=postgresql://myuser:mypassword@sl-db:5432/simplelogin

I already changed that to the database i made. And everytime after '@sl-db:' I would get this error. If i changed it to my server ip, local host.local ip same error just stating what i would enter instead of sl-db. 

I'm not the best in this, but it seems to me that the host is the problem for some reason. Which host name should i use

**Reply by joossii** [reply]
when i set it up without another postgres database running the confirmation email does not send. I get a 502 bad gateway

**Reply by nguyenkims** [reply]
> Which host name should i use

You can try 172.17.0.1 as it's often the host address.

> when i set it up without another postgres database running the confirmation email does not send. I get a 502 bad gateway

SimpleLogin can't work without the database.

**Reply by joossii** [reply]
i will try this but first theres another problem i need to fix see the other discussion
